core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -83,44 +78,35 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
         }
 
-        // parsed internally, but returned null during parsing...
         if (query == null) {
-            return null;
+            /*
+             * What the hack happened here? What we got from the user looks something like this:
+             *
+             * {
+             *   "query": {
+             *      "filtered": {
+             *          "query": { },
+             *          "filter": {"range": { "created": { "gte": "now - 1d / d" }}}
+             *      }
+             *   }
+             * }
+             * */
+            return EmptyQueryBuilder.PROTOTYPE;
         }
 
         if (filter == null) {
-            if (!filterFound) {
-                // we allow for null filter, so it makes compositions on the client side to be simpler
-                return query;
-            } else {
-                // even if the filter is not found, and its null, we should simply ignore it, and go
-                // by the query
-                return query;
-            }
-        }
-        if (Queries.isConstantMatchAllQuery(filter)) {
-            // this is an instance of match all filter, just execute the query
-            return query;
+            filter = EmptyQueryBuilder.PROTOTYPE;
```

Comment:
I am wondering whether we should just use null here, which would become EmptyQueryBuilder anyway in the builder